tfhe/src/core_crypto/commons/math/random/uniform.rs

```diff
@@ -1,11 +1,67 @@
 use super::*;
+use crate::core_crypto::algorithms::misc::divide_ceil;
+use crate::core_crypto::commons::numeric::UnsignedInteger;
 
 /// A distribution type representing uniform sampling for unsigned integer types. The value is
 /// uniformly sampled in `[0, 2^n[` where `n` is the size of the integer type.
 #[derive(Copy, Clone)]
 pub struct Uniform;
 
-macro_rules! implement_uniform {
+macro_rules! implement_uniform_uint {
+    ($T:ty) => {
+        impl RandomGenerable<Uniform> for $T {
+            type CustomModulus = $T;
+            #[allow(unused)]
+            fn generate_one<G: ByteRandomGenerator>(
+                generator: &mut RandomGenerator<G>,
+                distribution: Uniform,
+            ) -> Self {
+                let mut buf = [0; std::mem::size_of::<$T>()];
+                buf.iter_mut().for_each(|a| *a = generator.generate_next());
+                // We use from_le_bytes as most platforms are low endian, this avoids endianness
+                // issues
+                <$T>::from_le_bytes(buf)
```

Comment:
I dont really get the endianness comment, the random `buf` does not have an endianness attached to it, its just random bytes

Reply:
from_le_bytes, before it was a transmute a long time ago

Reply:
OK I get that, its just you could have used from_be_bytes, its would not have changed the randomness aspect since the buf is neither little or big endian, from_le_bytes advantage is just that since most platforms are little endian, it should be a no-op / simple mov

Reply:
Yes I think that’s what the comment says ?

Reply:
Or more clearly vs from native endian

Reply:
Its the `this avoids endianness issues` I don't agree with/understand. The `buf` has no endianness right ?

So using `from_ne_bytes` should be correct for example

Reply:
the idea is that we want randomness to work on all paltforms

the buffer has no endianness

if we do a transmute then we depend on the platform endianness (which used to be done a long time ago)

meaning for example that the seeded entities would be differently expanded depending on the platform

using either from_le_bytes or from_be_bytes is good as it avoids endianness issues that would happen with ne_bytes

Reply:
le bytes was preferred as it's one of the more commonly used ones as far as we could tell (btw the story of why intel CPUs are little endian are a very old implementation detail for some customer, I'll try to dig it up)

Reply:
https://stackoverflow.com/a/36263273/18088947

Reply:
Oh I see, I didn't think about the seeding thing

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/318